pkg/agent/util/syscall/syscall_windows_test.go

```diff
@@ -211,6 +213,100 @@ func TestListIPForwardRows(t *testing.T) {
 	assert.Equal(t, wantErr, gotErr)
 }
 
+func TestListIPForwardRowsSuccess(t *testing.T) {
+	// The table contains two rows. It will be returned by syscallN.
+	table := struct {
+		NumEntries uint32
+		Table      [2]MibIPForwardRow
+	}{
+		NumEntries: 2,
+		Table: [2]MibIPForwardRow{
+			{
+				Luid:  10,
+				Index: 11,
+				DestinationPrefix: AddressPrefix{
+					Prefix: RawSockAddrInet{
+						Family: AF_INET,
+						data:   [26]byte{10, 10, 10, 0},
+					},
+					prefixLength: 24,
+				},
+				NextHop: RawSockAddrInet{
+					Family: AF_INET,
+					data:   [26]byte{11, 11, 11, 11},
+				},
+			},
+			{
+				Luid:  20,
+				Index: 21,
+				DestinationPrefix: AddressPrefix{
+					Prefix: RawSockAddrInet{
+						Family: AF_INET,
+						data:   [26]byte{20, 20, 20, 0},
+					},
+					prefixLength: 24,
+				},
+				NextHop: RawSockAddrInet{
+					Family: AF_INET,
+					data:   [26]byte{21, 21, 21, 21},
+				},
+			},
+		},
+	}
+
+	syscallN := func(trap uintptr, args ...uintptr) (r1, r2 uintptr, err syscall.Errno) {
+		// This mocks the 1st call to syscallN made by getIPForwardTable. It makes args[1] point to the table.
+		if len(args) > 1 {
+			ptr := unsafe.Pointer(args[1])
```

Comment:
Somehow `checkptr` thinks "pointer arithmetic result points to invalid allocation".
There is no problem when `-race` is used. Need more investigation.